entity-management-definitions/src/main/java/eu/europeana/entitymanagement/utils/EntityObjectFactory.java

```diff
@@ -88,11 +88,35 @@
 
     try {
       @SuppressWarnings("unchecked")
-      Class<T> entityClass =
+      Class<T> consolidatedEntityClass =
           (Class<T>) consolidatedEntityTypesClassMap.get(EntityTypes.valueOf(entity.getType()));
-      // entityClass cannot be null here as map contains all possible types
-      return entityClass.getDeclaredConstructor(entityClass).newInstance(entity);
-
+      switch (EntityTypes.valueOf(entity.getType())) {
```

Comment:
It is required. In the previous case, when invoing entityClass.getDeclaredConstructor(entityClass).newInstance(entity), the constructor public Entity(Entity copy), was invoked, even though the entity objects were the derived ones like Organization, Place, etc. The argument to the constructor should be casted to Place, Organization, etc., so that the rigth constructor is invoked, like e.g. public Organization(Organization copy).